src/lib/minject/Injector.hx

```diff
@@ -350,6 +351,18 @@ class Injector
 	{
 		var info = new InjectorInfo(null, []);
 		addClassToInfo(forClass, info, []);
+		// sort rtti to ensure post constructors are last and in order
+		ArraySort.sort(info.fields, function(p1,p2) {
```

Comment:
I've used the stable `ArraySort(info.fields, fn)` here rather than `info.fields.sort(fn)`, because I was unsure if preserving the order of other fields matters.  If it doesn't, you could switch to `info.fields.sort` and I imagine have a more optimised implementation.